psi4/src/psi4/libsapt_solver/amplitudes.cc

```diff
@@ -620,7 +620,7 @@ void SAPT2::ijkl_to_ikjl(double *tARAR, int ilength, int jlength, int klength,
 
   for(int i=0; i<ilength; i++) {
     for(int l=0; l<llength; l++) {
-      long int ijkl = i*jlength*klength*llength + l;
+      long int ijkl = (long int)i*jlength*klength*llength + l;
```

Comment:
Nevermind, the compiler should promote all of the operands to long int before the multiplication takes place.